pkg/kv/iterators.go

```diff
@@ -252,25 +252,26 @@ type PartitionIterator struct {
 	err          error
 }
 
-func NewPartitionIterator(ctx context.Context, store Store, msgType protoreflect.MessageType, partitionKey string) (*PartitionIterator, error) {
-	itr, err := store.Scan(ctx, []byte(partitionKey), []byte(""))
-	if err != nil {
-		return nil, err
-	}
+func NewPartitionIterator(ctx context.Context, store Store, msgType protoreflect.MessageType, partitionKey string) *PartitionIterator {
```

Comment:
I think the issue with SeekGE is relevant also for KVOrderedCommitIterator

Reply:
That's a GC iterator, right? I don't assume it happens that often